ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -232,6 +217,90 @@ bool RSkComponentImage::processImageData(const char* path, char* response, int s
   return true;
 }
 
+inline void RSkComponentImage::drawContentShadow( SkCanvas *canvas,
+                            SkRect frameRect,
+                            SkRect targetRect,
+                            sk_sp<SkImage> imageData ,
+                            const ImageProps &imageProps){
+    if(!(layer()->componentShadow.isShadowVisible())){
+      return;
+    }
+
+    SkPaint shadowPaint;
+    setImageFilters(shadowPaint,imageProps,targetRect,frameRect,true,imageData->isOpaque());
+    bool SaveLAyerDone=false;
+    bool shadowOnFrame=(( frameRect.width() < targetRect.width()) || ( frameRect.height() < targetRect.height())||(imageProps.resizeMode == ImageResizeMode::Repeat));
+    SkMatrix identityMatrix;
+    SkRect shadowRect;
+    SkIRect imageFrameBounds;
+      //Calculate absolute frame bounds
+      if(shadowOnFrame) {
+        imageFrameBounds.setXYWH(frameRect.x() + layer()->componentShadow.shadowOffset.width(), frameRect.y() + layer()->componentShadow.shadowOffset.height(), frameRect.width(), frameRect.height());
+      } else {
+        imageFrameBounds.setXYWH(targetRect.x() + layer()->componentShadow.shadowOffset.width(), targetRect.y() + layer()->componentShadow.shadowOffset.height(), targetRect.width(), targetRect.height());
+     }
+    SkIRect shadowBounds=layer()->componentShadow.getShadowBounds(imageFrameBounds);
+    shadowRect=SkRect::Make(shadowBounds);
+
+    if(layer()->componentShadow.shadowOpacity) {
+      canvas->saveLayerAlpha(&shadowRect,layer()->componentShadow.shadowOpacity);
+      SaveLAyerDone=true;
+    }
+    if(!imageData->isOpaque() ) {
+      canvas->drawImageRect(imageData, targetRect, &shadowPaint);
+    } else {
+      if(!SaveLAyerDone) {
+        canvas->saveLayer(&shadowRect,&shadowPaint);
+      }
+      if(shadowOnFrame) {
+        canvas->clipRect(frameRect,SkClipOp::kDifference);
+      } else {
+        canvas->clipRect(targetRect,SkClipOp::kDifference);
+      }
+      shadowPaint.setColor(layer()->componentShadow.shadowColor);
+      canvas->drawIRect(imageFrameBounds, shadowPaint);
+    }
+    if(SaveLAyerDone) {
+      canvas->restore();
+    }
+    #ifdef SHOW_IMAGE_BOUND
+      SkPaint paint;
+      paint.setStyle(SkPaint::kStroke_Style);
+      paint.setColor(SK_ColorGREEN);
+      paint.setStrokeWidth(2);
+      canvas->drawRect(shadowRect,paint);
+    #endif
+}
+
+inline void RSkComponentImage::setImageFilters (SkPaint &paintObj,const ImageProps &imageProps,
+                                                      SkRect targetRect,SkRect frameRect ,
+                                                      bool setImageShadowFilter, bool isOpaque) {
```

Comment:
Can this be changed to filterForShadow to reflect the purpose & usage.
Currently setImageShadowFilter set as true for deawshadow & flase for DtawImage. Which contradicting the variable name